types/factory-girl/index.d.ts

```diff
@@ -12,11 +12,20 @@ declare const factory: factory.Static;
 declare namespace factory {
     type Generator<T> = () => T;
 
-    type Definition<T> = T | Generator<T>;
+    type Definition<T> = T | Generator<T> | Promise<T>;
```

Comment:
Each attribute can be `Promise<T>`.

- https://github.com/simonexmachina/factory-girl/blob/f7ec9d5004021ff1334ab9440e9b5cc4fd1a18b0/src/Factory.js#L43
- https://github.com/simonexmachina/factory-girl/blob/f7ec9d5004021ff1334ab9440e9b5cc4fd1a18b0/src/utils/asyncPopulate.js#L23